pallets/utility/src/lib.rs

```diff
@@ -218,7 +224,7 @@ decl_module! {
         /// If all were successful, then the `BatchCompleted` event is deposited.
         #[weight = batch_weight(&calls)]
         pub fn batch_atomic(origin, calls: Vec<<T as Trait>::Call>) {
-            let is_root = ensure_root(origin.clone()).is_ok();
+            let is_root = Self::is_root_with_permissions(origin.clone())?;
             Self::deposit_event(match with_transaction(|| {
                 for (index, call) in calls.into_iter().enumerate() {
```

Comment:
The call name, and module name in  `pallets/permissions` should be updated for every tx dispatched in the batch.